Plugins/Flow.Launcher.Plugin.PluginsManager/PluginsManager.cs

```diff
@@ -302,6 +310,62 @@ internal List<Result> Search(IEnumerable<Result> results, string searchName)
                 .ToList();
         }
 
+        internal List<Result> InstallFromWeb(string url)
+        {
+            var filename = url.Split("/").Last();
+            var name = filename.Split(string.Format(".{0}", zip)).First();
+
+            var plugin = new UserPlugin
+            {
+                ID = "",
+                Name = name,
+                Version = string.Empty,
+                Author = Context.API.GetTranslation("plugin_pluginsmanager_unknown_author"),
+                UrlDownload = url
+            };
+
+            var result = new Result
+            {
+                Title = string.Format(Context.API.GetTranslation("plugin_pluginsmanager_install_from_web"), filename),
+                SubTitle = plugin.UrlDownload,
+                IcoPath = icoPath,
+                Action = e =>
+                {
+                    if (e.SpecialKeyState.CtrlPressed)
+                    {
+                        SearchWeb.NewTabInBrowser(plugin.UrlDownload);
+                        return ShouldHideWindow;
+                    }
+
+                    if (Settings.WarnFromUnknownSource)
+                    {
+                        if (!InstallSourceKnown(plugin.UrlDownload)
+                            && MessageBox.Show(string.Format(Context.API.GetTranslation("plugin_pluginsmanager_install_unknown_source_warning"), 
+                                                        Environment.NewLine), 
+                                                Context.API.GetTranslation("plugin_pluginsmanager_install_unknown_source_warning_title"),
+                                                MessageBoxButton.YesNo) == MessageBoxResult.No)
+                            return false;
+                    }
+
+                    Application.Current.MainWindow.Hide();
+                    _ = InstallOrUpdate(plugin);
+                    
+                    return ShouldHideWindow;
+                }
+            };
+
+            return new List<Result> { result };
+        }
+
+        private bool InstallSourceKnown(string url)
+        {
+            var author = url.Split('/')[3];
+            var acceptedSource = "https://github.com";
+            var contructedUrlPart = string.Format("{0}/{1}/", acceptedSource, author);
+
+            return url.StartsWith(acceptedSource) && Context.API.GetAllPlugins().Any(x => x.Metadata.Website.StartsWith(contructedUrlPart));
+        }
```

Comment:
github is known source🧐🤔
Probably it will be better to treat all url download as unknown source

Reply:
github is only a known source if the author of the plugin is found in the pluginsmanifest, otherwise it's still treated as unknown source